src/arithmetic/arithmetic_expression_construct.c

```diff
@@ -411,31 +426,47 @@ static AR_ExpNode *_AR_ExpFromMapProjection(const cypher_astnode_t *expr) {
 			// { .name }
 			prop = cypher_ast_map_projection_property_get_prop_name(selector);
 			prop_name = cypher_ast_prop_name_get_value(prop);
-			children[i * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
+			tomapOp->op.children[j * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
 			AR_ExpNode *entity = AR_EXP_NewVariableOperandNode(entity_name);
-			children[i * 2 + 1] = AR_EXP_NewAttributeAccessNode(entity, prop_name);
+			tomapOp->op.children[j * 2 + 1] = AR_EXP_NewAttributeAccessNode(entity, prop_name);
+			j++;
 		} else if(t == CYPHER_AST_MAP_PROJECTION_LITERAL) {
 			// { v: n.v }
 			prop = cypher_ast_map_projection_literal_get_prop_name(selector);
 			prop_name = cypher_ast_prop_name_get_value(prop);
 			const cypher_astnode_t *literal_exp =
 				cypher_ast_map_projection_literal_get_expression(selector);
-			children[i * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
-			children[i * 2 + 1] = AR_EXP_FromASTNode(literal_exp);
+			tomapOp->op.children[j * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
+			tomapOp->op.children[j * 2 + 1] = AR_EXP_FromASTNode(literal_exp);
+			j++;
 		} else if(t == CYPHER_AST_MAP_PROJECTION_IDENTIFIER) {
 			// { v }
 			prop = cypher_ast_map_projection_identifier_get_identifier(selector);
 			prop_name = cypher_ast_identifier_get_name(prop);
-
-			children[i * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
-
-			children[i * 2 + 1] = AR_EXP_NewVariableOperandNode(prop_name);
+			tomapOp->op.children[j * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
+			tomapOp->op.children[j * 2 + 1] = AR_EXP_NewVariableOperandNode(prop_name);
+			j++;
+		} else if(t == CYPHER_AST_MAP_PROJECTION_ALL_PROPERTIES) {
+			// { .* }
+			// Use properties() to get a map with all properties
+			if(propertiesOp == NULL) {
+				propertiesOp = AR_EXP_NewOpNode("properties", false, 1);
+				propertiesOp->op.children[0] = AR_EXP_NewVariableOperandNode(entity_name);
+			}
 		} else {
 			ASSERT("Unexpected AST node type" && false);
 		}
 	}
 
-	return op;
+	if(propertiesOp) {
```

Comment:
I think queries such as: `MATCH (n) RETURN n{.*}` can be quite common, it is a bit of a shame to compute the merge between an empty map and the properties map when we can simply return the `properties` map without doing any extra work, i.e. `MATCH (n) RETURN n{.*}` becomes `MATCH (n) RETURN properties(n)` without the user even knowing about it.

Note this reopens the case where `tomapOp` can be NULL if `allProps_selectors == n_selectors`

Reply:
What do you think of return of ```(n_selectors == allProps_selectors)```?
```C
	if(allProps_selectors > 0) {
		// { .* }
		// Use properties() to get a map with all properties
		propertiesOp = AR_EXP_NewOpNode("properties", false, 1);
		propertiesOp->op.children[0] = AR_EXP_NewVariableOperandNode(entity_name);

		if(n_selectors == allProps_selectors) {
			return propertiesOp;
		}
	}
```
[096d24c](https://github.com/RedisGraph/RedisGraph/pull/2698/commits/096d24c65066c0130f67d0bbc805eb38d06ac31a)